src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -260,6 +282,50 @@ private async Task<Document> InlineTemporaryAsync(Document document, VariableDec
             return updatedDocument;
         }
 
+        private static bool MayContainSideEffects(SyntaxNode expression)
+        {
+            // Checks to see if inlining the temporary variable may change the code's semantics. 
+
+            if (expression == null)
+            {
+                return true;
+            }
+
+            if (expression.IsKind(SyntaxKind.ParenthesizedExpression, out ParenthesizedExpressionSyntax parenthesizedExpression))
+            {
+                expression = parenthesizedExpression.WalkDownParentheses();
+            }
```

Comment:
shouldn't be necessary.

Reply:
Wouldn't this be necessary for cases like:
```C#
class C
{
    int P { get; set; }

    void M()
    {
        var [||]c = (new C());
        c.P = 1;
        var c2 = c;
    }
}
```
?

Reply:
nope!  because you are calling DescendentNodesAndSelf below.  so you're already walking down the entire tree :)

Reply:
Ah, you're right. Thanks for the catch!